llvm/include/llvm/CAS/OnDiskHashMappedTrie.h

```diff
@@ -57,7 +57,7 @@ class OnDiskHashMappedTrie {
 
     explicit operator bool() const { return I == -2u; }
 
-    LookupResult() noexcept = default;
+    LookupResult() = default;
```

Comment:
It complain that the exception type of the constructor do not match (yes, it is build with -fno-exception). I don't know what is going on.

Reply:
Probably a bug related to the `Content` member, since `MappedContentReference` isn't marked `noexcept` (shouldn't matter, but...). Dropping it seems fine; frankly I'm not sure why it was here at all.

Reply:
```
/source/llvm/include/llvm/CAS/OnDiskHashMappedTrie.h: In member function 'llvm::cas::OnDiskHashMappedTrie::MappedContentReference llvm::cas::OnDiskHashMappedTrie::insert(llvm::ArrayRef<unsigned char>, llvm::StringRef, llvm::StringRef)':
/source/llvm/include/llvm/CAS/OnDiskHashMappedTrie.h:89:32: error: use of deleted function 'constexpr llvm::cas::OnDiskHashMappedTrie::LookupResult::LookupResult()'
     return insert(LookupResult(), Hash, Metadata, Data);
                                ^
/source/llvm/include/llvm/CAS/OnDiskHashMappedTrie.h:60:5: note: 'constexpr llvm::cas::OnDiskHashMappedTrie::LookupResult::LookupResult() noexcept' is implicitly deleted because its exception-specification does not match the implicit exception-specification ''
     LookupResult() noexcept = default;
     ^
```